linksight/api/models.py

```diff
@@ -63,72 +63,18 @@ class Match(models.Model):
     def __str__(self):
         return '{} ({})'.format(self.dataset.name, self.id)
 
-    def clean_matches(self, matches):
-        for col in matches:
-            if col.endswith('score'):
-                filler = 0
-            else:
-                filler = None
-            matches[col] = matches[col].where(pd.notnull(matches[col]),
-                                              filler)
-        return matches
-
-    def generate_match_items(self, **kwargs):
-        psgc = Dataset.objects.get(pk=settings.PSGC_DATASET_ID)
-        with psgc.file.open() as f:
-            psgc_df = pd.read_csv(f, dtype={'code': object})
-
-        with self.dataset.file.open() as f:
-            dataset_df = pd.read_csv(f)
-
-        psgc_df['province_code'] = (psgc_df['code']
-                                    .str.slice(stop=PROV_CODE_LEN)
-                                    .str.ljust(PSGC_LEN, '0'))
-        psgc_df['city_municipality_code'] = (psgc_df['code']
-                                             .str.slice(stop=CITY_MUN_CODE_LEN)
-                                             .str.ljust(PSGC_LEN, '0'))
-        interlevels = [
-            {
-                'name': 'province',
-                'dataset_field_name': kwargs.get('province_col'),
-                'reference_fields': ['Prov', 'Dist']
-            },
-            {
-                'name': 'city_municipality',
-                'dataset_field_name': kwargs.get('city_municipality_col'),
-                'reference_fields': ['City', 'Mun', 'SubMun']
-            },
-            {
-                'name': 'barangay',
-                'dataset_field_name': kwargs.get('barangay_col'),
-                'reference_fields': ['Bgy']
-            },
-        ]
-
-        matcher = LinkSightMatcher(dataset=dataset_df,
-                                   reference=psgc_df,
-                                   interlevels=interlevels)
-        matched_raw = matcher.get_matches()
-
-        matches = self.join_interlevels(matched_raw, dataset_df, interlevels)
-
-        matches = self._mark_matched(matches)
-
-        matches.rename(columns={'index': 'dataset_index'}, inplace=True)
-        matches = self._add_total_score(matches)
-
-        for _, row in matches.iterrows():
-            MatchItem.objects.create(match=self, **row.to_dict(), chosen=False)
-
     @staticmethod
     def _mark_matched(df):
         df['matched'] = ~df.duplicated('index', keep=False)
 
         def has_no_match(row):
-            if not row['matched_barangay'] and not row['matched_city_municipality'] and \
-                    not row['matched_province']:
-                return False
-            return row['matched']
+            if (
```

Comment:
So this will return null if there is no suitable match that meets the score requirement?